### PR TITLE
nix-prefetch: 0.4.0 -> 0.4.1

### PR DESCRIPTION
**Pull request by msteen**
###### Motivation for this change
I planned to finish a rewrite, but it has gotten too low on my priority to list, so I thought it better to merge the outstanding PRs and push a new version to include them. Of course I got the sha256 by running:
`nix-prefetch nix-prefetch` :wink: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/nix-prefetch/default.nix [hunk 1]
```diff
@@ -7,27 +7,19 @@ let
...
+    sha256 = "0ddpj0qykm8dvjndl9r187q5gwdir3silaik7qrwx9akmah4wi6b";
     # the stat call has to be in a subshell or we get the current date
     extraPostFetch = ''
       echo $(stat -c %Y $out) > $out/.timestamp
     '';
   };
 
-  patches = [
-    # Fix input_type regex for macOS systems: https://github.com/msteen/nix-prefetch/pull/24
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix-prefetch/default.nix:25]: Please remove the unused fetchpatch from inputs.

Comment (msteen) [reply pkgs/tools/package-management/nix-prefetch/default.nix:25]: I missed that, thanks! It is now removed.

**Comment by SuperSandro2000** [comment]
While I have you: can you add support for hash in vendor and fetchers? I recently had an issue with that in combination with nix-update.

**Comment by msteen** [comment]
@SuperSandro2000 I have updated it to include your patch, thanks!

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130508` at 05e6c5e5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nix-prefetch</li>
    <li>nix-update</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/package-management/nix-prefetch/default.nix:54:3:
```
   |
54 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/tools/package-management/nix-prefetch/default.nix:49:3:
```
   |
49 |   buildPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130508` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nix-prefetch</li>
    <li>nix-update</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130508` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nix-update</li>
  </ul>
</details>

**Comment by siraben** [comment]
Why doesn't `nix-prefetch` appear in PATH on macOS?

**Comment by msteen** [comment]
@siraben How am I supposed to know that? Maybe because it is a script, but many executables are that, so I doubt that would be the problem.

**Comment by msteen** [comment]
@siraben There was a commit added by @zowoq to change the meta to set platforms to all. I assume they did that to fix macOS. Unfortunately I have no experience with macOS, nor do I know how to test if it now works. Could you check if it is now alright?

PS: Did you manually add those `nix-preview` comments, or is that something automated that I could call upon too?

**Comment by zowoq** [comment]
Sorry, forgot to comment when I pushed the commit.

> Why doesn't `nix-prefetch` appear in PATH on macOS?

As `platforms` was unset it was only being built as a dependency of `nix-update` on darwin so `nixpkgs-review` didn't include it.

**Comment by siraben** [comment]
`nix-update` works on darwin now as a result of this PR! :tada:

**Pull request merged**